hathor/transaction/storage/migrations/migrate_static_metadata.py

```diff
@@ -0,0 +1,82 @@
+#  Copyright 2023 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import TYPE_CHECKING
+from unittest.mock import Mock
+
+from structlog import get_logger
+
+from hathor.conf.get_settings import get_global_settings
+from hathor.transaction import BaseTransaction
+from hathor.transaction.static_metadata import BlockStaticMetadata, TransactionStaticMetadata, VertexStaticMetadata
+from hathor.transaction.storage.migrations import BaseMigration
+from hathor.util import progress
+
+if TYPE_CHECKING:
+    from hathor.transaction.storage import TransactionStorage
+
+logger = get_logger()
+
+
+class Migration(BaseMigration):
+    def skip_empty_db(self) -> bool:
+        return True
+
+    def get_db_name(self) -> str:
+        return 'migrate_static_metadata'
+
+    def run(self, storage: 'TransactionStorage') -> None:
```

Comment:
Shouldn't we clear out the migrated fields in the previous metadata?

Reply:
Yes, however there's a problem that prevents us from doing it easily: the same method that is used for serialization, is used for API responses (`to_json`). As I had to keep the static attributes in that method to keep API compatibility, the serialized metadata also contains them.

To fix this we have to create separate methods for serialization and for APIs, but I would rather do this in a separate PR. Maybe I can just create an issue and do it next?

Reply:
I don't feel comfortable having duplicates (e.g. `meta.height` and `static_meta.height`). I guess we should migrate all attributes to the static metadata and even remove them from old metadata. For the `to_json()` method, I guess it will have to get information from the static metadata for now.

Reply:
Just to be clear: indeed the properties will be duplicate in the storage (temporarily), however in code the `metadata.create_from_json()` was updated so it does not include any of the migrated properties. This means the static metadata is the only source of truth, the only duplication is in `@properties` that are just forwards.

But if even then you feel it's better to fix this in this same PR, I could implement the change I mentioned above (separating a serialization method from the `to_json`) and merge it before this one, which will allow us to remove it completely from the old metadata.

Reply:
Fair, I'll create a new PR to be merged before this one that will allow me to fix this.

Reply:
I ended up doing it in this PR, in this new commit: https://github.com/HathorNetwork/hathor-core/pull/1014/commits/fa9c34ccfbe1ad2f449b108252dc92d3cac4e1c7. In it, I:

1. Created new `from_bytes()` and `to_bytes()` methods for `TransactionMetadata`, so we can use `to_json()` for APIs, with all attributes, and `to_bytes()` for the storage, without the attributes that were migrated.
2. Removed the `TransactionMetadata` aliases to static metadata attributes, and updated all usages (that is, for `height`, `min_height`, and `feature_activation_bit_counts`).
3. Updated the migration to remove migrated fields from metadata.

Reply:
Ok. Thanks!